common/configurable/src/main/java/io/helidon/common/configurable/Resource.java

```diff
@@ -163,35 +150,12 @@ static Resource from(InputStream inputStream, String description) {
      * @param keyPrefix prefix of keys that may contain the location of resource
      * @return a resource ready to load from one of the locations or empty if neither is defined
      */
-    static Optional<Resource> from(Config config, String keyPrefix) {
-        return OptionalHelper.from(config.get(keyPrefix + "-path")
-                                           .asOptionalString()
-                                           .map(Paths::get)
-                                           .map(Resource::from))
-                .or(() -> config.get(keyPrefix + "-resource-path")
-                        .asOptionalString()
-                        .map(Resource::from))
-                .or(() -> config.get(keyPrefix + "-url")
-                        .asOptional(URI.class)
-                        .map(uri -> config.get("proxy-host").value()
-                                .map(proxyHost -> {
-                                    if (config.get(keyPrefix + "-use-proxy").asBoolean(true)) {
-                                        Proxy proxy = new Proxy(Proxy.Type.HTTP,
-                                                                new InetSocketAddress(proxyHost,
-                                                                                      config.get("proxy-port").asInt(80)));
-                                        return Resource.from(uri, proxy);
-                                    } else {
-                                        return Resource.from(uri);
-                                    }
-                                })
-                                .orElseGet(() -> Resource.from(uri))))
-                .or(() -> config.get(keyPrefix + "-content-plain")
-                        .asOptionalString()
-                        .map(content -> Resource.fromContent("config:" + keyPrefix + "-content", content)))
-                .or(() -> config.get(keyPrefix + "-content")
-                        .asOptionalString()
-                        .map(Base64.getDecoder()::decode)
-                        .map(content -> Resource.fromContent("config:" + keyPrefix + "-content-b64", content)))
+    static Optional<Resource> create(Config config, String keyPrefix) {
```

Comment:
Why do we have to pass a key prefix ? Why can't the caller pass the right config node ?
This is introducing some level of abstraction that is doing magic.